app/javascript/components/automate-entry-points/index.jsx

```diff
@@ -0,0 +1,278 @@
+      setTimeout(() => {
+        setData((value) => updateTreeData(value, 'datastore_folder', newChildren));
+        resolve();
+      }, 1000);
+    })).then(() => {
+      setIsLoading(false);
+      miqSparkleOff();
+    });
+  }, []);
+
+  const onLoadData = ({ element }) => {
+    let path = element.name;
+    if (element.metadata && element.metadata.fqname) {
+      path = element.metadata.fqname;
+    }
+    API.get(`/api/automate/${path}?depth=1`).then((newNodes) => {
+      const newChildren = [];
+      newNodes.resources.forEach((newNode) => {
+        if (element.id !== newNode.id) {
+          if (newNode.klass !== 'MiqAeClass') {
+            newChildren.push({
+              id: newNode.id,
+              name: newNode.name,
+              children: [],
+              isBranch: true,
+              parent: element.id,
+              metadata: { domain_fqname: newNode.domain_fqname, fqname: newNode.fqname },
+            });
+          } else {
+            newChildren.push({
+              id: newNode.id,
+              name: newNode.name,
+              children: [],
+              parent: element.id,
+              metadata: { domain_fqname: newNode.domain_fqname, fqname: newNode.fqname },
+            });
+          }
+        }
+      });
+      return new Promise((resolve) => {
+        if (element.children.length > 0) {
+          resolve();
+          return;
+        }
+        setTimeout(() => {
+          setData((value) =>
+            updateTreeData(value, element.id, newChildren));
+          resolve();
+        }, 1000);
+      });
+    });
+  };
+
+  const wrappedOnLoadData = async({ element }) => {
+    const nodeHasNoChildData = element.children.length === 0;
+    const nodeHasAlreadyBeenLoaded = nodesAlreadyLoaded.find(
+      (e) => e.id === element.id
+    );
+
+    await onLoadData({ element });
+
+    if (nodeHasNoChildData && !nodeHasAlreadyBeenLoaded) {
+      setNodesAlreadyLoaded([...nodesAlreadyLoaded, element]);
+
+      setTimeout(() => {
+      }, 1000);
+    }
+  };
+  wrappedOnLoadData.propTypes = {
+    element: PropTypes.objectOf({ children: PropTypes.array, id: PropTypes.number }).isRequired,
+  };
+  wrappedOnLoadData.defaultProps = {
+  };
+
+  const onSelect = (value) => {
+    if (value.isBranch === false && value.isSelected) {
+      data.forEach((node) => {
+        if (selectedNode && (node.id === selectedNode.element.id)) {
+          document.getElementById(node.id).style.backgroundColor = 'transparent';
+        }
+      });
+      document.getElementById(value.element.id).style.backgroundColor = 'rgba(0, 0, 0, 0.2)';
+      setSelectedNode(value);
+      setDisableSubmit(false);
+    }
+  };
+
+  const onExpand = ((value) => {
+    if (value.isExpanded && selectedNode && document.getElementById(selectedNode.element.id)) {
+      document.getElementById(selectedNode.element.id).style.backgroundColor = 'rgba(0, 0, 0, 0.2)';
+    }
+  });
```

Comment:
Not important, but I noticed this has an unnecessary set of parens around it.
```suggestion
  const onExpand = (value) => {
    if (value.isExpanded && selectedNode && document.getElementById(selectedNode.element.id)) {
      document.getElementById(selectedNode.element.id).style.backgroundColor = 'rgba(0, 0, 0, 0.2)';
    }
  };
```